src/SonarScanner.Shim/PathHelper.cs

```diff
@@ -22,40 +22,86 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using SonarQube.Common;
 
 namespace SonarScanner.Shim
 {
     public static class PathHelper
     {
-        public static bool IsPartOfAProject(string filePath, IEnumerable<string> projectPaths)
+        public static string WithTrailingDirectorySeparator(this DirectoryInfo directory)
         {
-            return projectPaths.Any(projectPath => IsInFolder(filePath, projectPath));
+            if (directory == null)
+            {
+                throw new ArgumentNullException(nameof(directory));
+            }
+
+            if (directory.FullName.EndsWith(Path.DirectorySeparatorChar.ToString()) ||
+                directory.FullName.EndsWith(Path.AltDirectorySeparatorChar.ToString()))
+            {
+                return directory.FullName;
+            }
+
+            if (directory.FullName.Contains(Path.AltDirectorySeparatorChar))
+            {
+                return directory.FullName + Path.AltDirectorySeparatorChar;
+            }
+
+            return directory.FullName + Path.DirectorySeparatorChar;
         }
 
-        public static bool IsInFolder(string filePath, string folder)
+        public static bool IsInDirectory(FileInfo file, DirectoryInfo directory)
         {
-            var normalizedPath = Path.GetDirectoryName(Path.GetFullPath(filePath));
-            return normalizedPath.StartsWith(folder, StringComparison.OrdinalIgnoreCase);
+            var normalizedDirectoryPath = directory.WithTrailingDirectorySeparator();
+            return file.FullName.StartsWith(normalizedDirectoryPath, FileInfoEqualityComparer.ComparisonType);
```

Comment:
This won't work if the directory and file are using different separators (e.g. if the project file contains _sub\foo.cs_ and _sub/bar.cs_ then presumably one of them won't match what we calculate the base directory to be).

Reply:
As far as I have seen the `DirectoryInfo` and `FileInfo` classes fix the separator so the comparison should work.

Reply:
I am adding tests.